### ISSUE TITLE
Runtime panics in 1.49.0-nightly

### ISSUE THREAD

**Issue by antonok-edm** [brave/adblock-rust#147]
Currently `lifeguard` is using a language feature that has been deprecated since Rust 1.39, which is technically a source of undefined behavior.

This is detected at runtime as of 1.49.0-nightly, which causes runtime panics during network rule matching:
```
'attempted to leave type `std::mem::ManuallyDrop<std::vec::Vec<u64>>` uninitialized, which is invalid'
```

`lifeguard` should be updated as soon as a version with the fix is published.

Relevant `lifeguard` issue: https://github.com/zslayton/lifeguard/issues/25
